libs-scala/executors/src/main/scala/com/daml/InstrumentedExecutors.scala

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.executors
+
+import java.util.concurrent.{ExecutorService, ThreadFactory, Executors => JavaExecutors}
+
+import com.codahale.metrics.{InstrumentedExecutorService, MetricRegistry}
+import com.daml.metrics.ExecutorServiceMetrics
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutorService}
+
+object InstrumentedExecutors {
+
+  def newWorkStealingExecutor(
+      name: String,
+      parallelism: Int,
+      registry: MetricRegistry,
+      executorServiceMetrics: ExecutorServiceMetrics,
+      errorReporter: Throwable => Unit = ExecutionContext.defaultReporter,
+  ): ExecutionContextExecutorService = {
+    val executorService = JavaExecutors.newWorkStealingPool(parallelism)
+    val executorServiceWithMetrics =
+      addMetricsToExecutorService(name, registry, executorServiceMetrics, executorService)
+    ExecutionContext.fromExecutorService(executorServiceWithMetrics, errorReporter)
+  }
+
+  def newFixedThreadPool(
+      name: String,
+      nThreads: Int,
```

Comment:
```suggestion
      numberOfThreads: Int,
```

Reply:
I wouldn't change this because it's in sync with how it's declared in the executor itself and in the java constructor